crowbar_framework/app/models/service_object.rb

```diff
@@ -682,9 +682,43 @@ def validate_proposal proposal
   #
   def validate_proposal_after_save proposal
     validate_proposal_constraints proposal
+    validate_postponed_nodes proposal
     handle_validation_errors
   end
 
+  #
+  # Make sure that when user wants to apply changes affecting the nodes that are postponed
+  # (= not upgraded while rest of the cloud is already upgraded)
+  # the nodes can be skipped.
+  #
+  def validate_postponed_nodes(proposal)
+    return unless upgrade_postponed?
+
```

Comment:
I would streamline and exit fast from here if there is nothing to do:

```
skip_unready_nodes = Rails.application.config.experimental.fetch("skip_unready_nodes", {})

skip_unready_nodes_enabled = skip_unready_nodes.fetch("enabled", false)
raise Crowbar::NeedOptionEnabled("skip_unready_nodes") unless skip_unready_nodes_enabled

skip_unready_nodes_roles = skip_unready_nodes.fetch("roles", [])
return if skip_unready_nodes_roles.size == 0

```

Reply:
> raise Crowbar::NeedOptionEnabled("skip_unready_nodes") unless skip_unready_nodes_enabled

Hm, there's no reason to raise here until we find out if any postponed node has a role in this barclamp.

> return if skip_unready_nodes_roles.size == 0

Why would we do it? Again, I think we must compare `skip_unready_nodes_roles` with the roles being applied.